src/ui/interaction.rs

```diff
@@ -17,6 +25,13 @@ pub struct InteractionPalette {
     pub pressed: Color,
 }
 
+/// Component that calls a [one-shot system](https://bevyengine.org/news/bevy-0-12/#one-shot-systems)
+/// when the [`Interaction`] component on the same entity changes to [`Interaction::Pressed`].
+/// Use this in conjuction with [`Commands::register_one_shot_system`] to create a callback for e.g. a button press.
+#[derive(Component, Debug, Reflect, Deref, DerefMut)]
+#[reflect(Component, from_reflect = false)]
+pub struct OnPress(#[reflect(ignore)] pub SystemId);
+
```

Comment:
IMO it's nice to have the related system immediately after the component, so I'd move this to right after `apply_interaction_palette` and move `apply_interaction_callback` up to right before `trigger_interaction_sfx`.

Also, IMO rename `apply_interaction_callback` -> `apply_on_press`, since that's the only interaction callback we have atm.

Reply:
[`f78efe9` (#229)](https://github.com/TheBevyFlock/bevy_quickstart/pull/229/commits/f78efe9a6337539ae78e1d6448f4794650aeda9c)

Moved system up instead to keep `pub` items above private ones.

Reply:
I see, I didn't realize we were doing that. Well, as long as it's consistent.